app/api_topologies.go

```diff
@@ -553,9 +553,9 @@ func (r *Registry) RendererForTopology(topologyID string, values url.Values, rpt
 		}
 	}
 	if len(filters) > 0 {
-		return topology.renderer, render.ComposeFilterFuncs(filters...), nil
+		return topology.renderer, render.Transformers([]render.Transformer{render.ComposeFilterFuncs(filters...), topology.filter}), nil
```

Comment:
Could we just cons up a list of `Transformer`s instead of creating `filters` then turning that into a `Transformer`?

Reply:
That wouldn't be desirable. We have
```go
type FilterFunc func(report.Node) bool

type Transformer interface {
	Transform(nodes Nodes) Nodes
}
```
 `FilterFunc`s are `Transformer`s. Invoking their `Transform` method calls `FilterFunc` for every node in `nodes` and constructs a result from the nodes for which that predicate is true. Adjacencies of the surviving nodes are pruned accordingly.

`ComposeFilterFuncs` yields another `FilterFunc`. Crucially that means invoking `Transform` on that is still only making a single pass over `nodes`, and is only constructing a single result, and only prunes adjacencies once.

By contrast, composing the same `FilterFunc`s as `Transformer`s would entail a full `nodes` pass, result construction and pruning for every `FilterFunc`.